### PR TITLE
[TRAFODION-2129] Trafodion to avoid use of deprecated HBase APIs/Classes

### PR DESCRIPTION
**Pull request by selvaganesang**
References to the deprecated classes HConnection, HConnectionManager and
HBaseAdmin are reworked to use Connection, ConnectionFactory and Admin classes
respectively.

A connection object is created and it is passed around to ensure that the
same connection object is used within the same process. Connection object is thread safe.

Admin object is created and closed after its use immediately. Creation of admin object is lightweight and the object is not thread safe

Also fixed the random hang at the time of rollback transaction waiting for the response from the TM process. TM process was hanging because it assumed that the request from RS didn't get responded when there was a UnknownTransactionException. This caused the test to hang and failed PR tests in the earlier attempts.

### CODE REVIEW

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/SsccTransactionalTable.java [hunk 1]
```diff
@@ -130,12 +129,6 @@
 
     static {
         config.set("hbase.hregion.impl", "org.apache.hadoop.hbase.regionserver.transactional.TransactionalRegion");
```
Comment (narendragoyal) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/SsccTransactionalTable.java]: Perhaps, this line is also not needed.

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionalTable.java [hunk 2]
```diff
@@ -99,7 +98,7 @@
  */
 public class TransactionalTable extends HTable implements TransactionalTableClient {
     static final Log LOG = LogFactory.getLog(RMInterface.class);
-    static private HConnection connection = null;
+    static private Connection connection = null;
     static Configuration       config = HBaseConfiguration.create();
     static ExecutorService     threadPool;
```
Comment (narendragoyal) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionalTable.java]: Please check - Dont think we need the threadPool object any more.

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/SsccTransactionalTable.java (hunk 1 above)
Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/SsccTransactionalTable.java]: Done.

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionalTable.java (hunk 2 above)
Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionalTable.java]: Tried. But, It looks like there is no HTable constructor that takes byte[] and Connection.

**Comment by selvaganesang** [comment]
Jenkins, extra tests

**Comment by selvaganesang** [comment]
Jenkins, extra tests

**Comment by selvaganesang** [comment]
Jenkins, extra tests

**Comment by DaveBirdsall** [comment]
Looks good to me.

**Comment by trinakrug** [comment]
Looks good.

**Comment by sbroeder** [comment]
Good to me.

**Comment by selvaganesang** [comment]
From: Selva Govindarajan [mailto:selva.govindarajan@esgyn.com] 
Sent: Tuesday, August 2, 2016 8:07 AM
To: Sean Broeder <sean.broeder@esgyn.com>; Trina Krug <trina.krug@esgyn.com>; Narendra Goyal <narendra.goyal@esgyn.com>; Shang-Sheng Tung <shang-sheng.tung@esgyn.com>; Prashanth Vasudev <prashanth.vasudev@esgyn.com>
Subject: RE: Help with PR 626 https://github.com/apache/incubator-trafodion/pull/627
 
I am sorry. It is 627. 
 
https://github.com/apache/incubator-trafodion/pull/627
 
 
From: Sean Broeder [mailto:sean.broeder@esgyn.com] 
Sent: Tuesday, August 2, 2016 12:26 AM
To: Selva Govindarajan <selva.govindarajan@esgyn.com>; Trina Krug <trina.krug@esgyn.com>; Narendra Goyal <narendra.goyal@esgyn.com>; Shang-Sheng Tung <shang-sheng.tung@esgyn.com>; Prashanth Vasudev <prashanth.vasudev@esgyn.com>
Subject: RE: Help with PR 626 https://github.com/apache/incubator-trafodion/pull/626
 
Selva,
Is this the correct pull request?  It seems to deal with update statistics and I don’t see any TM code changes.
 
Thanks,
Sean

From: Selva Govindarajan [mailto:selva.govindarajan@esgyn.com] 
Sent: Friday, July 29, 2016 12:11 PM
To: Trina Krug <trina.krug@esgyn.com>; Sean Broeder <sean.broeder@esgyn.com>; Narendra Goyal <narendra.goyal@esgyn.com>; Shang-Sheng Tung <shang-sheng.tung@esgyn.com>
Subject: RE: Help with PR 626 https://github.com/apache/incubator-trafodion/pull/626
 
I tried with the change to set the request pending count. It worked.  But, in the log I didn’t see any “Unknown Transaction “ during abort.  Is there a way to force it during abort to check if it works. 
 
UnknownTransaction” during commit seems to work ok and there was no hang without this change too.
 
Selva

From: Sean Broeder [mailto:sean.broeder@esgyn.com] 
Sent: Friday, July 29, 2016 9:32 AM
To: Selva Govindarajan <selva.govindarajan@esgyn.com>; Narendra Goyal <narendra.goyal@esgyn.com>; Shang-Sheng Tung <shang-sheng.tung@esgyn.com>; Trina Krug <trina.krug@esgyn.com>
Subject: RE: Help with PR 626 https://github.com/apache/incubator-trafodion/pull/626
 
Hi Selva,
On abort transaction if we get an unknown transaction exception I think we need to decrement the request pending count.  We sent the message and got a reply in the form of an exception, so we are not waiting on an additional reply, but if the count never gets to zero we’ll hang.
 
Regards,
Sean
 
From: Selva Govindarajan [mailto:selva.govindarajan@esgyn.com] 
Sent: Friday, July 29, 2016 9:23 AM
To: Narendra Goyal <narendra.goyal@esgyn.com>; Sean Broeder <sean.broeder@esgyn.com>; Shang-Sheng Tung <shang-sheng.tung@esgyn.com>; Trina Krug <trina.krug@esgyn.com>
Subject: RE: Help with PR 626 https://github.com/apache/incubator-trafodion/pull/626
 
On further discussion with Narendra, it is found that 
 
Trafodion.dtm.log shows that the transaction abort returned unknownTransaction from RS
 
2016-07-29 08:25:07,561 ERROR transactional.TransactionManager: Abort of transaction: 94547 participantNum: 2 region: TRAFODION.S025.T025T005,,1469780693295.8b5acee4643097d33435d2250e6a7d3b. threw Exception: java.io.IOException: UnknownTransactionException
 
This UnknownTransactionException is the DoNotRetryIOException. It looks like this exception may not be completing the request and hence it is possible that we are hung. 
 
I am retaining my workspace in this state so that we can debug further if we need to.
 
Selva

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java [hunk 3]
```diff
@@ -2436,56 +2409,44 @@ public Integer call() throws IOException {
...
-            catch(Exception e){
-                LOG.error("FATAL Exception calling abortDDL for transaction: " + transactionState.getTransactionId() + "Exception: "  + e);
-                throw new UnsuccessfulDDLException(e);
-            }
+                loopExit = true; 
+              } 
+              catch (InterruptedException ie) {}
+              catch (CommitUnsuccessfulException cue) {
+                 loopExit = true;
+                 LOG.error("Exception at the time of aborting DDL transaction", cue);
```
Comment (prashanth-vasudev) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java]: AbortDDL is not started yet here. The error message needs to get updated something like "CommitUnSuccessful Exception received from one or more of abort threads before servicing abortDDL"

Comment (selvaganesang) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java]: Will do next time when this area of code is modified.  This change is already merged

**Comment by prashanth-vasudev** [comment]
Rest of the code looks good to me. Thanks.

**Pull request closed without merging**